processor/src/decoder/trace.rs

```diff
@@ -299,8 +300,27 @@ impl DecoderTrace {
         self.hasher_trace[0].push(group_ops_left);
         self.hasher_trace[1].push(parent_addr);
 
-        for idx in USER_OP_HELPERS {
-            self.hasher_trace[idx].push(ZERO);
+        match op {
+            // The Emit operation pushes the event id in hasher_trace[2]. This value is picked up by
+            // the Op Group table.
+            Operation::Emit(event_id) => {
+                let mut idx_iter = USER_OP_HELPERS.clone();
+
+                // Set hasher_trace[2] = event_id
+                let emit_imm_value_idx = idx_iter.next().unwrap();
+                assert_eq!(emit_imm_value_idx, EMIT_IMM_VALUE_HASHER_TRACE_IDX);
+                self.hasher_trace[emit_imm_value_idx].push(event_id.into());
+
+                // Set the other hasher trace columns to 0
+                for idx in idx_iter {
+                    self.hasher_trace[idx].push(ZERO);
+                }
+            },
+            _ => {
+                for idx in USER_OP_HELPERS {
+                    self.hasher_trace[idx].push(ZERO);
+                }
+            },
```

Comment:
Why do we need a special handler for this here instead of moving this logic into the `op_emit()` handler we have in the `operations` module?

We could use `self.decoder.set_user_op_helpers()` there to set the value of the helper registers. This would be more consistent with how other operations work (e.g., u32 operations).